### ISSUE TITLE
Only one (non randomly) hiking trail is displayed on shared sections

### ISSUE THREAD

**Issue by gorn** [osmandapp/Osmand#1721]
When there are multiple hiking trails along one road (and we have red, green, blue and yellow trails in Czech republic), only one of them is displayed. Moreover the trail which "wins" is chosen somehow randomly, it even does not respect the importance of the trail (if given by tags like network=lwn). 

It is very confusing - trails appear and disappear and there is no way to find out where particular trail goes if it is hidden. I understand that it might be difficult to make it as on paper maps (where all colours as countour are visible), but current situation is nearly the worst solution. There are several intermediate solutions - for example display trails symbols on shared sections (little bit something like http://hiking.waymarkedtrails.org/cs/?zoom=16&lat=49.56641&lon=13.78198&hill=0 does - not ideal, but at least contains the information). Or just a way to "select" or "highlight" some trail so it shows it in its entirety.

**Comment by xmd5a2** [comment]
Yes, I know about this problem. But our current rendering engines can't do this. For now name and ref are displayed from all relations on shared sections.
![_2015-09-21_09-39-45](https://cloud.githubusercontent.com/assets/5829014/9986449/781afebe-6045-11e5-8c5d-822b9b5d1e05.png)
Also that's why I introduced "Coloring according to route scope" option (may be it's not the best name) (walkingRoutesScopeOSMC) which renders hiking trails according to network tag.
![_2015-09-21_09-44-21](https://cloud.githubusercontent.com/assets/5829014/9986470/bd9c6edc-6045-11e5-8fdd-2ed195e9ce04.png)

**Comment by vshcherb** [comment]
That's known issue and for now quite difficult to fix.

**Comment by gorn** [comment]
I understand that is is difficult to render it perfectly (more paraller color lines), but the problem now is that it is *impossibble* to find out what trail is on given segment. Yes, there are refs rendered (thanks for it of course!) but, very often they are not visible on the shared segment. If might be sufficient if for example shared segments are coloured brown and the colours would be only shown as the symbol. If it is impossible to have more symbols on one segment, than it would be possible to use "combo" symbols like this one:

![this one](http://img.radio.cz/pictures/r/turistika/turisticka_znacka3.jpg)

**Comment by asyschikov** [comment]
Are there any updates on this issue? I am using OsmAnd extensively for hiking, it is a great app! Unfortunately in trail dense areas it is hard to follow a specific trail. If this issue is too hard to fix, is it possible to highlight a specific trail? Or alternatively force app to follow a specific trail when using walking navigation? Thanks!

**Comment by vshcherb** [comment]
Yes we are targetting to display lines of trails cause rendering is even more complicated to fix. We hope to be able to select and show trails in the next release

**Comment by m93a** [comment]
FYI here's how Mapy.cz (free offline navigation app that is very popular in Czechia) do it:

![55811487_424954011598090_8232679053404930048_n](https://user-images.githubusercontent.com/1671665/55294833-e29d3000-5406-11e9-8e06-5c51673c8922.png)

The colored lines next to paths and roads indicate the trail color. This way it's very easy to keep track of the correct trail without cluttering the map with icons.

**Comment by scaidermern** [comment]
So unfortunately rendering of overlapping hiking routes is still impossible.

As an improvement would it be possible to render at least multiple route symbols next to each other? Example:

![photo_2019-05-03_11-25-54](https://user-images.githubusercontent.com/1040784/57130444-bf64f800-6d99-11e9-98a2-cf486bef9719.jpg)

Is this possible with the current rendering engine?

**Comment by gorn** [comment]
There is also a complementary possibility - allow users to highlight a touristic route somehow, simiraly to bus route. It is not complete solution, but it would be a great improvement against current state. Currently not only I do not see it the trail goes somewhere but in case I think it does I have NO WAY to check whther this is true.

**Comment by scaidermern** [comment]
> There is also a complementary possibility - allow users to highlight a touristic route somehow, simiraly to bus route. It is not complete solution, but it would be a great improvement against current state.

Indeed! There is already a feature request: #3969

**Comment by vshcherb** [comment]
@scaidermern it is not possible today and there is no ideas how to make it yet

**Comment by Gollam** [comment]
@vshcherb I have a related question. If only one can be drawn, is there currently a precedence between iwn, nwn, rwn and lwn paths? Because as I see, iwn is almost always chosen for example in my area, instead of nwn, which makes using the map more confusing, because the signs painted in the forest are for the nwn path.

**Comment by vshcherb** [comment]
We've introduced order 
```

**Comment by vshcherb** [comment]
we think to stick to that order for now until we can't draw multiple segments

**Comment by Gollam** [comment]
@vshcherb ok, that sounds good enough. I see that ordering was introduced in the OsmAnd-tools repo in March. Should I already see this order of precedence in my area with the latest vector map? I'm just asking to confirm whether I'm seeing a bug or not, because iwn is still preferred against nwn here.

**Comment by vshcherb** [comment]
Another issue was opened for 3.9 to make iwn routes optional cause they could be non-sense for some local usages

**Comment by Mashin6** [comment]
@vshcherb Wouldn't a solution be that the colored lines would not be computed in the smartphone app, but they would be pre-calculated on your side as new line objects and they would be provided as a separated data layer (same as contour lines and hillshade are now)?
You would then only have to build a pipeline that would make those objects out of relations and apply offset if there are multiple routes on one path segment.

**Comment by vshcherb** [comment]
I think this issue will be moved to selecting trails on the map which will partially fix the issue.

Obviously displaying multiple symbols on the line is great though it could be up to 10 symbols on the line, so technically it will be difficult to do. Even more important selecting "relation routes" on the map has higher priority
